### ISSUE TITLE
i don't understand 

### ISSUE THREAD

**Issue by manhcntt21** [Isminoula/TextNormSeq2Seq#2]
https://github.com/Isminoula/TextNormSeq2Seq/blob/2f3207efc47121200410e357b503261c9de5d25c/lib/data/DataLoader.py#L97

https://github.com/Isminoula/TextNormSeq2Seq/blob/2f3207efc47121200410e357b503261c9de5d25c/lib/data/DataLoader.py#L109
you can explain for me, difference between line97 and line 109

As far as I understand, are NOISE  cases (Training character-based  model) also added to word_tweets, or how.

Tks

**Comment by Isminoula** [comment]
Sorry for the confusion on the "word" here, this is meant to indicate that we are training on _word pairs_ (the character level model). Line 97 adds each source and target pair of words in the list of training examples, as it is, i.e. original version from training data. On the other hand, line 109 adds word pairs with injected noise.

**Comment by manhcntt21** [comment]
ok, I understand that, but during the debugging process, I found NOISE were not added, I had to convert word_tweets.append(tweet) 
 to word_tweets.append(copy.deepcopy(tweet)) at line 109  for that to be done.

**Comment by Isminoula** [comment]
Can you share the command you used to run the character level model?

**Comment by manhcntt21** [comment]
I use the command you suggested, 

python main.py -logfolder -save_dir spelling_model -gpu 0 -input spelling -data_augm -noise_ratio 0.1 -attention -bias -lowercase -bos -eos -brnn -batch_size 500 -dropout 0.5 -emb_size 256 -end_epoch 50 -layers 3 -learning_rate_decay 0.05 -lr 0.001 -max_grad_norm 5 -rnn_size 500 -rnn_type 'LSTM'  -tie_decoder_embeddings -share_embeddings -share_vocab -start_decay_after 30 -teacher_forcing_ratio 0.6  -max_train_decode_len 50

**Comment by manhcntt21** [comment]
affter i  edit comment at line 109 : into 
word_tweets.append(copy.deepcopy(tweet)) 

I found that word_tweets added noise

**Comment by Isminoula** [comment]
Ok let me check, maybe I mistyped something when I was cleaning up the code before uploading. I'll run it again and let you know soon!

**Comment by manhcntt21** [comment]
Thank you for your enthusiasm and friendliness

**Comment by Isminoula** [comment]
You are right about deepcopy, thanks for catching that! The repo is updated :)

**Comment by manhcntt21** [comment]
OK

**Comment by Isminoula** [comment]
Closing this ticket as it was resolved. Thanks!
